components/ErrorEventsDashboard.js

```diff
@@ -1,134 +1,90 @@
 import moment from 'moment'
-import { Component } from 'react'
-import { Button } from 'react-bootstrap'
-import { withAuth } from 'use-auth0-hooks'
+import useSWR from 'swr'
 
-import { secureFetch } from '../util/middleware'
+import FetchMessage from './FetchMessage'
 
-class ErrorEventsDashboard extends Component {
-  constructor (props) {
-    super(props)
-    this.state = {
-      events: null,
-      eventsError: null
-    }
-  }
-
-  handleFetchErrors = async (force = false) => {
-    const { events, eventsError } = this.state
-    if (!force && (events || eventsError)) {
-      return
-    }
-
-    const { accessToken } = this.props.auth
-    if (!accessToken) {
-      return
-    }
-    const url = `${process.env.API_BASE_URL}/api/admin/bugsnag/eventsummary`
-    const fetchedErrorEvents = await secureFetch(url, accessToken)
-    if (fetchedErrorEvents.status === 'sucess') {
-      this.setState({
-        events: fetchedErrorEvents.data,
-        fetchMessage: `Updated at ${moment().format('h:mm:ss a')}`
-      })
-    } else {
-      this.setState({
-        fetchMessage: `Failed to update!`
-      })
-    }
-  }
-
-  async componentDidMount () {
-    await this.handleFetchErrors()
-  }
-
-  render () {
-    const { auth } = this.props
-    const { events, eventsError, fetchMessage } = this.state
-    if (!auth.isAuthenticated) return null
-    const hasEvents = events && events.length > 0
-    return (
-      <div>
-        <h2>Error Events Summary</h2>
-        <div className='controls'>
-          <Button onClick={this.handleFetchErrors}>
-            Fetch errors
-          </Button>
```

Comment:
This button to refetch is gone. Granted, with SWR, it gets automatically refetched every 30 seconds, but would it be useful to have a way to refetch immediately or at least show that refetching will occur in x# of seconds?

Reply:
Resolved in https://github.com/ibi-group/otp-admin-ui/pull/17/commits/7c774280b1d29ef63708133ecd38062515fb33ee (added auto fetch message and button, which I think makes for a nice approach).